### ISSUE TITLE
Additional search fields

### ISSUE THREAD

**Issue by stefan-franz** [mixxxdj/mixxx#11824]
### Feature Description

E.g. i want to search tracks with more search operations like: location: Fox/German bpm:>120 comment:party 
it would be very useful, to have 2-4 additional search fields with a pulldown menu where i can sele the search operations as witten above. And additional to search for a artist e.g. "Jürgen Drews"
So every additional Option a pulldown Menü e.g location and next a field where i can type something. 
Hope i described that to understand. 

And great would of course filter folder to save in a structure like Playlists "Filter Folders" with a complete query as above.

**Comment by nikkuAg** [comment]
Is this issue still open? I want to work on it.

**Comment by ronso0** [comment]
FYI the legacy Mixxx interface engine will be switched to QML in one of the next releases. There have been preparational steps and we'll continue to work on that after 2.4 has been released. There was also already a library redesign branch (stale as it had some serious performance regressions) which was post-poned to the QML refactoring. Additionally, the library backend is unfortunately coupled to the GUI, so this also needs to be rewritten for QML.

That being said, and considering implementing such a feature is a lot of work and it has to be re-implemented in QML, and it's not clear if the effort is worth it.

If you want to start working on it regardless, I suggest you take a look at the `WSearchLineEdit` class. The query parsing is done in `SearchQueryParser`/`SearchQuery`.

**Comment by nikkuAg** [comment]
Is the new release QML open source? Can I contribute on it?

**Comment by Swiftb0y** [comment]
Yes, just build the main branch with `-DQML=ON -DQT6=ON` to try it out (currently very far from finished and incomplete, but you can load tracks by drag-and-dropping on a player).

**Comment by nikkuAg** [comment]
Okay I will work on it

**Comment by ronso0** [comment]
Noteworthy I think: the QML table #4594

**Comment by Holzhaus** [comment]
> > Yes, just build the main branch with `-DQML=ON -DQT6=ON` to try it out (currently very far from finished and incomplete, but you can load tracks by drag-and-dropping on a player).
> 
> Sorry, but I was not able to understand your point on where to add -DQML=ON -DQT6=ON. I ran the command in the readme stating `cmake ..` and `cmake --build .`. I was trying to add the flag in the second command, but it says -dqt6 argument not found. Is there anything specific I am missing?

You need to add the flags to the first command 😉

**Comment by nikkuAg** [comment]
![image](https://github.com/mixxxdj/mixxx/assets/77184239/62ad05ce-a12a-45bd-b2f3-3dce1b1c0db3)

Is there any specific change required to solve this error?

**Comment by daschuer** [comment]
This idicates that the Qt 6 libraries are not found. 
Are you on MacOS or windows? Than you can use the vcpkg environment from the top of the mixxxdj/vcpkg 2.5-rel branch. That comes with Qt6. It has not been merged yet because some issues. So maybe you need also merge:
https://github.com/mixxxdj/mixxx/pull/11701

**Comment by nikkuAg** [comment]
> 

I am using wsl2 no windows 11

**Comment by nikkuAg** [comment]
![image](https://github.com/mixxxdj/mixxx/assets/77184239/09f6a421-6239-4603-9ad7-63ebfd24179b)
After installing Qt6 via apt and running the build command I faced this error

**Comment by daschuer** [comment]
did you install all the packages by this command: 
https://github.com/mixxxdj/mixxx/blob/main/tools/debian_buildenv_qt6.sh#L48

**Comment by nikkuAg** [comment]
build was successful, but I was not able to see how to make changes and run the development mode of mixx. Is it possible in wsl or should I switch to windows

**Comment by nikkuAg** [comment]
I appreciate your help in setting up Mixx. But now I need help understanding open mixx as a development mode. Is this possible in wsl or not?

**Comment by daschuer** [comment]
I have no experience. Did the build succeed now. 
Than you can start Mixxx via 
`mixxx --developer`

**Comment by ronso0** [comment]
@nikkuAg Github Issues are for tracking bugs and feature requests, for development support there's https://mixxx.zulipchat.com/#narrow/stream/247620-development-help
Please also search existing topics there first.
Thank you!
